test/parallel/test-find-package-json.js

```diff
@@ -0,0 +1,155 @@
+    );
+  });
+
+  it('should accept a file URL instance', () => {
+    const fixturesDirUrl = `${pathToFileURL(fixtures.fixturesDir)}`;
+    const importMetaUrl = `${fixturesDirUrl}/whatever.ext`;
+    const specifierBase = './packages/root-types-field/';
+    assert.strictEqual(
+      findPackageJSON(
+        new URL('index.js', `${fixturesDirUrl}${specifierBase.slice(1)}`),
+        new URL(importMetaUrl),
+      ),
+      fixtures.path(specifierBase, 'package.json'),
+    );
+  });
+
+  it('should be able to crawl up (CJS)', () => {
+    const pathToMod = fixtures.path('packages/nested/sub-pkg-cjs/index.js');
+    const parentPkg = require(pathToMod);
+
+    const pathToParent = fixtures.path('packages/nested/package.json');
+    assert.strictEqual(parentPkg, pathToParent);
+  });
+
+  it('should be able to crawl up (ESM)', () => {
+    const pathToMod = fixtures.path('packages/nested/sub-pkg-esm/index.js');
+    const parentPkg = require(pathToMod).default; // This test is a CJS file
+
+    const pathToParent = fixtures.path('packages/nested/package.json');
+    assert.strictEqual(parentPkg, pathToParent);
+  });
+
+  it('can require via package.json', () => {
+    const pathToMod = fixtures.path('packages/cjs-main-no-index/other.js');
+    // `require()` falls back to package.json values like "main" to resolve when there is no index
+    const answer = require(pathToMod);
+
+    assert.strictEqual(answer, 43);
+  });
+
+  it('should be able to do whatever the heck Antoine created', () => {
+    tmpdir.refresh();
+
+    fs.writeFileSync(tmpdir.resolve('entry.mjs'), `
+      import { findPackageJSON } from 'node:module';
+      import fs from 'node:fs';
+
+      const readPJSON = (locus) => JSON.parse(fs.readFileSync(locus, 'utf8'));
+
+      const { secretNumberRoot } = readPJSON(findPackageJSON('pkg', import.meta.url));
+      let secretNumberSubfolder; // Impossible to resolve because pkg's pjson.exports blocks access
+      try { ({ secretNumberSubfolder } = findPackageJSON('pkg/subfolder/', import.meta.url)) } catch {}
+      const { secretNumberSubfolder2 } = readPJSON(findPackageJSON('pkg2', import.meta.url));
+      const { secretNumberPkg2 } = readPJSON(findPackageJSON('pkg2/package.json', import.meta.url));
+
+      console.log(secretNumberRoot, secretNumberSubfolder, secretNumberSubfolder2, secretNumberPkg2);
+    `);
+
+    const secretNumberRoot = Math.ceil(Math.random() * 999);
+    const secretNumberSubfolder = Math.ceil(Math.random() * 999);
+    const secretNumberSubfolder2 = Math.ceil(Math.random() * 999);
+    const secretNumberPkg2 = Math.ceil(Math.random() * 999);
+
+    fs.mkdirSync(tmpdir.resolve('node_modules/pkg/subfolder'), { recursive: true });
+    fs.writeFileSync(
+      tmpdir.resolve('node_modules/pkg/subfolder/index.js'),
+      '',
+    );
+    fs.writeFileSync(
+      tmpdir.resolve('node_modules/pkg/subfolder/package.json'),
+      JSON.stringify({
+        type: 'module',
+        secretNumberRoot,
```

Comment:
```suggestion
        secretNumberSubfolder,
```

Reply:
Sure! Are these all aligned? If so, please do click the button (and kindly restart ci so it's done in the morning).